src/systemctl/systemctl.c

```diff
@@ -7517,8 +7569,17 @@ static int find_paths_to_edit(sd_bus *bus, char **names, char ***paths) {
 
                         if (arg_full)
                                 r = unit_file_create_copy(&lp, unit_name, path, &new_path, &tmp_path);
-                        else
-                                r = unit_file_create_new(&lp, unit_name, ".d/override.conf", &new_path, &tmp_path);
+                        else {
+                                _cleanup_free_ char *path_copy = strdup(path);
```

Comment:
none of the codepaths will actually make use of the `_cleanup_` handler of this, hence drop it?